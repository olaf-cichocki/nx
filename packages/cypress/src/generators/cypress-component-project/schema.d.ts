export interface CypressComponentProjectSchema {
  project: string;
  componentType: 'react' | 'next';
  compiler: 'swc' | 'babel';
  force?: boolean;
}
