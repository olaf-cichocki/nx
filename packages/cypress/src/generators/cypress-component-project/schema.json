{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxCypressComponentProject",
  "cli": "nx",
  "title": "Set up Cypress component test for a project",
  "type": "object",
  "examples": [
    {
      "command": "nx g @nrwl/cypress:cypress-component-project --project=my-cool-lib --componentType=react",
      "description": "Add cypress component testing to an existing project named my-cool-lib"
    },
    {
      "command": "nx g @nrwl/cypress:cypress-component-project --project=my-cool-lib --componentType=react --force",
      "description": "Add Cypress component testing to an existing project that already has a Cypress component testing enabled. Helpful to reset the configuration to the default."
    }
  ],
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project to convert",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What project should we add Cypress component testing to?"
    },
    "componentType": {
      "type": "string",
      "enum": ["react", "next"],
      "description": "Which framework is being used?",
      "alias": "framework",
      "x-prompt": "What framework is being used in this project?"
    },
    "compiler": {
      "type": "string",
      "enum": ["swc", "babel"],
      "description": "Which compiler is being used in the existing project?",
      "default": "babel"
    },
    "force": {
      "type": "boolean",
      "description": "Overwrite an existing cypress component test project. This is helpful if you want to reset to the default configurations.",
      "default": false
    }
  },
  "required": ["project", "componentType"]
}
