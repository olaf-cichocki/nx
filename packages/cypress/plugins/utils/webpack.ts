import { Configuration } from 'webpack';
import { TsconfigPathsPlugin } from 'tsconfig-paths-webpack-plugin';
import { posix } from 'path';
import { getHashDigest, interpolateName } from 'loader-utils';

function getCSSModuleLocalIdent(context, localIdentName, localName, options) {
  // Use the filename or folder name,
  // based on some uses the index.js / index.module.(css|scss|sass) project style
  const fileNameOrFolder = context.resourcePath.match(
    /index\.module\.(css|scss|sass|styl)$/
  )
    ? '[folder]'
    : '[name]';
  // Create a hash based on the file location and class name.
  // Will be unique across a project, and close to globally unique.
  const hash = getHashDigest(
    posix.relative(context.rootContext, context.resourcePath) + localName,
    'md5',
    'base64',
    5
  );
  // Use loaderUtils to find the file or folder name
  const className = interpolateName(
    context,
    `${fileNameOrFolder}_${localName}__${hash}`,
    options
  );
  // Remove the .module that appears in every classname
  // when based on the file and replace all "." with "_".
  return className.replace('.module_', '_').replace(/\./g, '_');
}

const loaderModulesOptions = {
  modules: {
    mode: 'local',
    getLocalIdent: getCSSModuleLocalIdent,
  },
  importLoaders: 1,
};

const commonLoaders = [
  {
    loader: require.resolve('style-loader'),
  },
  {
    loader: require.resolve('css-loader'),
    options: loaderModulesOptions,
  },
];

export const CSS_MODULES_LOADER = {
  // TODO(caleb): does css outside of css-in-js work?
  test: /\.css$|\.scss$|\.sass$|\.less$|\.styl$/,
  oneOf: [
    {
      test: /\.module\.css$/,
      use: commonLoaders,
    },
    {
      test: /\.module\.(scss|sass)$/,
      use: [
        ...commonLoaders,
        {
          loader: require.resolve('sass-loader'),
          options: {
            implementation: require('sass'),
            sassOptions: {
              fiber: false,
              precision: 8,
            },
          },
        },
      ],
    },
    {
      test: /\.module\.less$/,
      use: [
        ...commonLoaders,
        {
          loader: require.resolve('less-loader'),
        },
      ],
    },
    {
      test: /\.module\.styl$/,
      use: [
        ...commonLoaders,
        {
          loader: require.resolve('stylus-loader'),
        },
      ],
    },
  ],
};

export function getLibLoaders(
  compiler: 'babel' | 'swc',
  preset: 'react' | 'next'
) {
  if (compiler === 'swc') {
    return {
      test: /\.([jt])sx?$/,
      loader: require.resolve('swc-loader'),
      exclude: /node_modules/,
      options: {
        jsc: {
          parser: {
            syntax: 'typescript',
            decorators: true,
            tsx: true,
          },
          transform: {
            react: {
              runtime: 'automatic',
            },
          },
          loose: true,
        },
      },
    };
  }

  return {
    test: /\.(js|jsx|mjs|ts|tsx)$/,
    loader: require.resolve('babel-loader'),
    options: {
      presets: [`@nrwl/${preset}/babel`],
      rootMode: 'upward',
      babelrc: true,
    },
  };
}

export function buildBaseWebpackConfig({
  tsConfigPath = 'tsconfig.cy.json',
  compiler = 'babel',
  preset = 'react',
}: {
  tsConfigPath: string;
  compiler: 'swc' | 'babel';
  preset: 'react' | 'next';
}): Configuration {
  const extensions = ['.ts', '.tsx', '.mjs', '.js', '.jsx'];
  return {
    target: 'web',
    resolve: {
      extensions,
      plugins: [
        new TsconfigPathsPlugin({
          configFile: tsConfigPath,
          extensions,
        }) as never,
      ],
    },
    mode: 'development',
    devtool: false,
    output: {
      publicPath: '/',
      chunkFilename: '[name].bundle.js',
    },
    module: {
      rules: [
        {
          test: /\.(bmp|png|jpe?g|gif|webp|avif)$/,
          type: 'asset',
          parser: {
            dataUrlCondition: {
              maxSize: 10_000, // 10 kB
            },
          },
        },
        CSS_MODULES_LOADER,
        getLibLoaders(compiler, preset),
      ],
    },
  };
}
