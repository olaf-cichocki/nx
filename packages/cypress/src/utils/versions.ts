export const nxVersion = '*';
export const eslintPluginCypressVersion = '^2.10.3';
/**
 * TODO(caleb): this is only the mac version, and temp until there is a v10 release
 */
export const cypressVersion =
  'https://cdn.cypress.io/beta/npm/10.0.0/darwin-x64/10.0-release-8765e6481bbc3b8247bd6f64139b8d3676d84c89/cypress.tgz';
export const cypressReactVersion = '^5.12.1';
export const cypressWebpackVersion = '^1.8.0';
export const webpackHttpPluginVersion = '^5.5.0';
export const swcLoaderVersion = '0.1.15';
export const swcCoreVersion = '~1.2.143';
