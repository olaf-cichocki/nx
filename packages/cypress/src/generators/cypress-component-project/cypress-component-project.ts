import { CypressComponentProjectSchema } from './schema';
import {
  addDependenciesToPackageJson,
  formatFiles,
  generateFiles,
  joinPathFragments,
  offsetFromRoot,
  ProjectConfiguration,
  readProjectConfiguration,
  Tree,
  updateJson,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import {
  cypressReactVersion,
  cypressVersion,
  cypressWebpackVersion,
  swcCoreVersion,
  swcLoaderVersion,
  webpackHttpPluginVersion,
} from '../../utils/versions';

import { CYPRESS_COMPONENT_TEST_TARGET } from '../../utils/project-name';
import {
  ComponentTestingProjectState,
  cypressComponentTestingState,
} from '../utils/verify-cypress-component-project';

export async function cypressComponentProject(
  tree: Tree,
  options: CypressComponentProjectSchema
) {
  const projectConfig = readProjectConfiguration(tree, options.project);

  const cypressState = cypressComponentTestingState(tree, projectConfig, {
    force: options.force,
  });

  let installDepsTask;
  switch (cypressState) {
    case ComponentTestingProjectState.INSTALL:
      installDepsTask = updateDeps(tree, options, true);
      break;
    case ComponentTestingProjectState.NO_INSTALL:
      installDepsTask = updateDeps(tree, options, false);
      break;
    case ComponentTestingProjectState.ALREADY_SETUP:
      throw new Error(
        'The project already has a cypress component testing target. Please use the --force flag to overwrite the existing project.'
      );
    case ComponentTestingProjectState.UPGRADE:
      throw new Error(
        'Cypress version of 10 or higher is required to create a component testing project'
      );
    default:
      throw new Error(
        `Unable to determine if project is able to use Cypress component testing`
      );
  }

  addProjectFiles(tree, projectConfig, options);
  updateTSConfig(tree, projectConfig, options);
  addTargetToProject(tree, projectConfig, options);

  return () => {
    formatFiles(tree);
    installDepsTask();
  };
}

function updateDeps(
  tree: Tree,
  options: CypressComponentProjectSchema,
  shouldInstallCypress: boolean
) {
  const devDeps = {
    '@cypress/webpack-dev-server': cypressWebpackVersion,
    'html-webpack-plugin': webpackHttpPluginVersion,
  };

  if (shouldInstallCypress) {
    devDeps['cypress'] = cypressVersion;
  }

  if (options.componentType === 'react' || options.componentType === 'next') {
    devDeps['@cypress/react'] = cypressReactVersion;
    if (options.compiler === 'swc') {
      devDeps['@swc/core'] = swcCoreVersion;
      devDeps['swc-loader'] = swcLoaderVersion;
    }
  }
  return addDependenciesToPackageJson(tree, {}, devDeps);
}

function updateGitIgnore(tree: Tree) {
  let ignoreContents;

  if (tree.exists('.gitignore')) {
    const contents = tree.read('.gitignore', 'utf-8');

    if (contents.includes('.next')) {
      return;
    }
    ignoreContents = `${contents}\n.next`;
  } else {
    ignoreContents = '.next';
  }
  tree.write('.gitignore', ignoreContents);
}

function addProjectFiles(
  tree: Tree,
  projectConfig: ProjectConfiguration,
  options: CypressComponentProjectSchema
) {
  // prevent overwriting the existing next.config.js
  const nextConfigPath = joinPathFragments(
    projectConfig.root,
    'next.config.js'
  );
  if (tree.exists(nextConfigPath)) {
    tree.rename(nextConfigPath, `${nextConfigPath}.bak`);
    updateGitIgnore(tree);
  }

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    projectConfig.root,
    {
      ...options,
      projectRoot: projectConfig.root,
      offsetFromRoot: offsetFromRoot(projectConfig.root),
      ext: '',
    }
  );

  if (options.componentType !== 'next') {
    tree.delete(joinPathFragments(projectConfig.root, 'next.config.js'));
  } else {
    updateGitIgnore(tree);
  }

  if (tree.exists(`${nextConfigPath}.bak`)) {
    tree.rename(`${nextConfigPath}.bak`, nextConfigPath);
  }
}

function updateTSConfig(
  tree: Tree,
  projectConfig: ProjectConfiguration,
  options: CypressComponentProjectSchema
) {
  const projectTsConfigPath = joinPathFragments(
    projectConfig.root,
    'tsconfig.json'
  );
  if (!tree.exists(projectTsConfigPath)) {
    throw new Error(
      `Expected project tsconfig.json to exist. Please create one. Expected ${projectTsConfigPath} to exist. Found none.`
    );
  }

  updateJson(tree, projectTsConfigPath, (json) => {
    json.references = json.references || [];
    json.references.push({ path: './tsconfig.cy.json' });
    return json;
  });

  // ignore the cypress files otherwise the build will fail
  // attempt to grab the tsconfig used for building
  const buildTsConfig = getTsConfig(projectConfig, options);

  if (buildTsConfig) {
    updateJson(tree, buildTsConfig, (json) => {
      json.exclude = Array.from(
        new Set([
          ...(json.exclude || []),
          'cypress/**/*',
          'cypress.config.ts',
          '**/*.cy.ts',
          '**/*.cy.js',
          '**/*.cy.tsx',
          '**/*.cy.jsx',
        ])
      );
      return json;
    });
  }
}

function getTsConfig(
  projectConfig: ProjectConfiguration,
  options: CypressComponentProjectSchema
): string {
  if (projectConfig.targets?.['build']?.options?.tsConfig) {
    return projectConfig.targets['build'].options.tsConfig;
  }

  if (projectConfig.projectType === 'library') {
    return joinPathFragments(projectConfig.root, 'tsconfig.lib.json');
  }

  if (
    projectConfig.projectType === 'application' &&
    options.componentType === 'next'
  ) {
    return joinPathFragments(projectConfig.root, 'tsconfig.json');
  }

  if (
    projectConfig.projectType === 'application' &&
    options.componentType === 'react'
  ) {
    return joinPathFragments(projectConfig.root, 'tsconfig.app.json');
  }

  // fallback to default tsconfig as that should always be there.
  return joinPathFragments(projectConfig.root, 'tsconfig.json');
}

function addTargetToProject(
  tree: Tree,
  projectConfig: ProjectConfiguration,
  options: CypressComponentProjectSchema
) {
  projectConfig.targets[CYPRESS_COMPONENT_TEST_TARGET] = {
    executor: '@nrwl/cypress:cypress',
    options: {
      cypressConfig: joinPathFragments(projectConfig.root, 'cypress.config.ts'),
      testingType: 'component',
    },
  };

  updateProjectConfiguration(tree, options.project, projectConfig);
}
