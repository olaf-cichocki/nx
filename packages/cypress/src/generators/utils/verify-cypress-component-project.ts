import { joinPathFragments, ProjectConfiguration, Tree } from '@nrwl/devkit';
import { CYPRESS_COMPONENT_TEST_TARGET } from '../../utils/project-name';
import { installedCypressVersion } from '../../utils/cypress-version';

export enum ComponentTestingProjectState {
  /**
   * the project already has component testing setup.
   * but did not provide a force flag to override the setup.
   */
  ALREADY_SETUP,
  /**
   * Safe to install cypress.
   */
  INSTALL,
  /**
   * A cypress version > 10 is already installed. No need to install cypress.
   */
  NO_INSTALL,
  /**
   * A cypress version < 10 is installed. Need to upgrade to v10 before allowing a cypress component project to be created.
   */
  UPGRADE,
}

/**
 * Check the installed version of cypress to determine
 * if a project is compatible with cypress component testing.
 */
export function cypressComponentTestingState(
  tree: Tree,
  projectConfig: ProjectConfiguration,
  options: InstallOptions = { force: false }
): ComponentTestingProjectState {
  // if the project already has the component testing target or a cypress.config.ts file then we make sure the --force flag is sent. if not, we throw an error.
  if (
    projectConfig.targets?.[CYPRESS_COMPONENT_TEST_TARGET] ||
    tree.exists(joinPathFragments(projectConfig.root, 'cypress.config.ts'))
  ) {
    if (!options.force) {
      return ComponentTestingProjectState.ALREADY_SETUP;
    }
    return ComponentTestingProjectState.INSTALL;
  }

  const installedVersion = installedCypressVersion();
  // if we don't have a version then it's safe to install cypress
  if (!installedVersion) {
    return ComponentTestingProjectState.INSTALL;
  }

  if (installedVersion >= 10) {
    return ComponentTestingProjectState.NO_INSTALL;
  }

  return ComponentTestingProjectState.UPGRADE;
}

export interface InstallOptions {
  force: boolean;
}
