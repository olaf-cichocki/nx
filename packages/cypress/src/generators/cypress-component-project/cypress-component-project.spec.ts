import {
  addProjectConfiguration,
  ProjectConfiguration,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { cypressComponentProject } from './cypress-component-project';
import { CYPRESS_COMPONENT_TEST_TARGET } from '../../utils/project-name';
import {
  ComponentTestingProjectState,
  cypressComponentTestingState,
} from '../utils/verify-cypress-component-project';

jest.mock('../utils/verify-cypress-component-project');
let projectConfig: ProjectConfiguration = {
  projectType: 'library',
  sourceRoot: 'libs/cool-lib/src',
  root: 'libs/cool-lib',
  targets: {
    build: {
      executor: '@nrwl/web:rollup',
      options: {
        tsConfig: 'libs/cool-lib/tsconfig.lib.json',
      },
    },
    test: {
      executor: '@nrwl/jest:jest',
      options: {
        jestConfig: 'libs/cool-lib/jest.config.js',
      },
    },
  },
};
describe('Cypress Component Project', () => {
  let tree: Tree;
  let mockedProjectState: jest.Mock<
    ReturnType<typeof cypressComponentTestingState>
  > = cypressComponentTestingState as never;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    addProjectConfiguration(tree, 'cool-lib', projectConfig);
    tree.write(
      '.gitignore',
      `
# compiled output
/dist
/tmp
/out-tsc
    `
    );
    tree.write(
      'libs/cool-lib/tsconfig.lib.json',
      `
{
  "extends": "./tsconfig.json",
  "exclude": [
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.spec.tsx",
    "**/*.test.tsx",
    "**/*.spec.js",
    "**/*.test.js",
    "**/*.spec.jsx",
    "**/*.test.jsx",
    "**/*.cy.ts"
  ],
  "include": [
    "**/*.js",
    "**/*.jsx",
    "**/*.ts",
    "**/*.tsx"
  ]
}
`
    );
    tree.write(
      'libs/cool-lib/tsconfig.json',
      `
{
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
`
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('validation', () => {
    it('should not be able to create project with < cypress v9 installed', async () => {
      mockedProjectState.mockReturnValue(ComponentTestingProjectState.UPGRADE);
      try {
        await cypressComponentProject(tree, {
          project: 'cool-lib',
          compiler: 'swc',
          componentType: 'react',
          force: false,
        });
        // make sure we reject to prevent the test from passing if an error isn't thrown
        return Promise.reject(
          'cypressComponentProject should have throw an error when cypress version requires an upgrade'
        );
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it('should throw an error if cypress project is already created', async () => {
      mockedProjectState.mockReturnValue(
        ComponentTestingProjectState.ALREADY_SETUP
      );
      tree.write('libs/cool-lib/cypress.config.ts', '');
      try {
        await cypressComponentProject(tree, {
          project: 'cool-lib',
          compiler: 'swc',
          componentType: 'react',
          force: false,
        });
        return Promise.reject(
          'cypressComponentProject should have throw an error when cypress project is already created'
        );
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it('should create cypress project over existing with --force', async () => {
      mockedProjectState.mockReturnValue(ComponentTestingProjectState.INSTALL);
      tree.write('libs/cool-lib/cypress.config.ts', '');
      const newTarget = {
        [CYPRESS_COMPONENT_TEST_TARGET]: {
          executor: '@nrwl/cypress:cypress',
          options: {
            cypressConfig: 'libs/cool-lib/cypress.config.ts',
            testingType: 'component',
          },
        },
      };
      updateProjectConfiguration(tree, 'cool-lib', {
        ...projectConfig,
        targets: {
          ...projectConfig.targets,
          ...newTarget,
        },
      });

      await cypressComponentProject(tree, {
        project: 'cool-lib',
        compiler: 'swc',
        componentType: 'react',
        force: true,
      });
      const actualProjectConfig = readProjectConfiguration(tree, 'cool-lib');

      expect(tree.exists('libs/cool-lib/cypress.config.ts')).toEqual(true);
      expect(tree.exists('libs/cool-lib/cypress')).toEqual(true);
      expect(tree.exists('libs/cool-lib/tsconfig.cy.json')).toEqual(true);
      expect(
        actualProjectConfig.targets[CYPRESS_COMPONENT_TEST_TARGET]
      ).toMatchSnapshot();
    });
  });

  it('should update build tsconfig', async () => {
    mockedProjectState.mockReturnValue(ComponentTestingProjectState.INSTALL);
    await cypressComponentProject(tree, {
      project: 'cool-lib',
      compiler: 'swc',
      componentType: 'react',
      force: false,
    });

    expect(
      tree.read('libs/cool-lib/tsconfig.lib.json', 'utf-8')
    ).toMatchSnapshot();
  });

  ['react', 'next'].forEach((framework: 'next' | 'react') => {
    describe(framework, () => {
      function assertProjectFiles() {
        if (framework === 'next') {
          expect(tree.exists('libs/cool-lib/next.config.js')).toEqual(true);
          expect(tree.read('.gitignore', 'utf-8')).toContain('.next');
        }
        expect(tree.exists('libs/cool-lib/cypress.config.ts')).toEqual(true);
        expect(tree.exists('libs/cool-lib/cypress')).toEqual(true);
        expect(tree.exists('libs/cool-lib/tsconfig.cy.json')).toEqual(true);
      }

      beforeEach(() => {
        mockedProjectState.mockReturnValue(
          ComponentTestingProjectState.INSTALL
        );
      });

      it(`should create project w/babel`, async () => {
        tree.delete('libs/cool-lib/cypress.config.ts');
        await cypressComponentProject(tree, {
          project: 'cool-lib',
          compiler: 'babel',
          componentType: framework,
          force: false,
        });

        const actualProjectConfig = readProjectConfiguration(tree, 'cool-lib');
        assertProjectFiles();
        expect(
          actualProjectConfig.targets[CYPRESS_COMPONENT_TEST_TARGET]
        ).toMatchSnapshot();
      });

      it('should create project w/swc', async () => {
        tree.delete('libs/cool-lib/cypress.config.ts');
        await cypressComponentProject(tree, {
          project: 'cool-lib',
          compiler: 'swc',
          componentType: framework,
          force: false,
        });

        const actualProjectConfig = readProjectConfiguration(tree, 'cool-lib');

        assertProjectFiles();
        expect(tree.exists('libs/cool-lib/tsconfig.cy.json')).toEqual(true);
        expect(
          actualProjectConfig.targets[CYPRESS_COMPONENT_TEST_TARGET]
        ).toMatchSnapshot();
      });
    });
  });
});
