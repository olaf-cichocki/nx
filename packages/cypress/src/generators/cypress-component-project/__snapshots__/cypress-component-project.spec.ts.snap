// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cypress Component Project next should create project w/babel 1`] = `
Object {
  "executor": "@nrwl/cypress:cypress",
  "options": Object {
    "cypressConfig": "libs/cool-lib/cypress.config.ts",
    "testingType": "component",
  },
}
`;

exports[`Cypress Component Project next should create project w/swc 1`] = `
Object {
  "executor": "@nrwl/cypress:cypress",
  "options": Object {
    "cypressConfig": "libs/cool-lib/cypress.config.ts",
    "testingType": "component",
  },
}
`;

exports[`Cypress Component Project react should create project w/babel 1`] = `
Object {
  "executor": "@nrwl/cypress:cypress",
  "options": Object {
    "cypressConfig": "libs/cool-lib/cypress.config.ts",
    "testingType": "component",
  },
}
`;

exports[`Cypress Component Project react should create project w/swc 1`] = `
Object {
  "executor": "@nrwl/cypress:cypress",
  "options": Object {
    "cypressConfig": "libs/cool-lib/cypress.config.ts",
    "testingType": "component",
  },
}
`;

exports[`Cypress Component Project should update build tsconfig 1`] = `
"{
  \\"extends\\": \\"./tsconfig.json\\",
  \\"exclude\\": [
    \\"**/*.spec.ts\\",
    \\"**/*.test.ts\\",
    \\"**/*.spec.tsx\\",
    \\"**/*.test.tsx\\",
    \\"**/*.spec.js\\",
    \\"**/*.test.js\\",
    \\"**/*.spec.jsx\\",
    \\"**/*.test.jsx\\",
    \\"**/*.cy.ts\\",
    \\"cypress/**/*\\",
    \\"cypress.config.ts\\",
    \\"**/*.cy.js\\",
    \\"**/*.cy.tsx\\",
    \\"**/*.cy.jsx\\"
  ],
  \\"include\\": [
    \\"**/*.js\\",
    \\"**/*.jsx\\",
    \\"**/*.ts\\",
    \\"**/*.tsx\\"
  ]
}
"
`;

exports[`Cypress Component Project validation should create cypress project over existing with --force 1`] = `
Object {
  "executor": "@nrwl/cypress:cypress",
  "options": Object {
    "cypressConfig": "libs/cool-lib/cypress.config.ts",
    "testingType": "component",
  },
}
`;

exports[`Cypress Component Project validation should not be able to create project with < cypress v9 installed 1`] = `[Error: Cypress version of 10 or higher is required to create a component testing project]`;

exports[`Cypress Component Project validation should throw an error if cypress project is already created 1`] = `[Error: The project already has a cypress component testing target. Please use the --force flag to overwrite the existing project.]`;
